engine/core/src/main/java/org/datacleaner/util/LabelUtils.java

```diff
@@ -96,13 +97,26 @@ public static String getLabel(ComponentJob job) {
      * 
      * @return
      */
-    public static String getLabel(ComponentJob job, boolean includeDescriptorName, boolean includeInputColumnNames,
+    public static String getLabel(ComponentJob job, boolean includeDescriptorName, boolean includeInputColumnNames, boolean includeRequirements) {
+        return getLabel(job, null, includeDescriptorName, includeInputColumnNames, includeRequirements);
+    }
+
+    public static String getLabel(ComponentJob job, AnalyzerResult result, boolean includeDescriptorName, boolean includeInputColumnNames,
             boolean includeRequirements) {
         final String jobName = job.getName();
         final StringBuilder label = new StringBuilder();
         if (Strings.isNullOrEmpty(jobName)) {
             ComponentDescriptor<?> descriptor = job.getDescriptor();
-            label.append(descriptor.getDisplayName());
+            String displayName = descriptor.getDisplayName();
+            if (result != null) {
+                if (result instanceof HasLabelAdvice) {
```

Comment:
Normally HasLabelAdvice is applied to the component, not the result. Do we need to change that? I worry if it will create confusion about which label will be used - the component's label or the result's label.

If we decide to also support it on the result class, then we should at least update the HasLabelAdvice javadoc accordingly.